pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_sql_statistics_persisted failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_sql_statistics_persisted [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9217669?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9217669?buildTab=artifacts#/) on release-23.1 @ [aa0b102c7aec3a68ec0617e67841ff58c0e24080](https://github.com/cockroachdb/cockroach/commits/aa0b102c7aec3a68ec0617e67841ff58c0e24080):


```
=== RUN   TestExecBuild_sql_statistics_persisted
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9986f69167dd2633633c749cd542ec4c/logTestExecBuild_sql_statistics_persisted49840248
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:2841: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5631/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted:73: EXPLAIN (VERBOSE)
        SELECT
          aggregated_ts,
          fingerprint_id,
          app_name,
          node_id,
          agg_interval,
          metadata,
          statistics
        FROM
          system.transaction_statistics
        expected:
              distribution: local
              vectorized: true
            ·
              • scan
                columns: (aggregated_ts, fingerprint_id, app_name, node_id, agg_interval, metadata, statistics)
                estimated row count: 1,000 (missing stats)
                table: transaction_statistics@primary
                spans: FULL SCAN
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • scan
              columns: (aggregated_ts, fingerprint_id, app_name, node_id, agg_interval, metadata, statistics)
              estimated row count: 1 (100% of the table; stats collected <hidden> ago)
              table: transaction_statistics@primary
              spans: FULL SCAN
[16:49:32] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5631/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted with config local: 3 tests, 1 failures
    logic.go:3942: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5631/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted:95: error while processing
    logic.go:3942: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5631/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted:95: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9986f69167dd2633633c749cd542ec4c/logTestExecBuild_sql_statistics_persisted49840248
--- FAIL: TestExecBuild_sql_statistics_persisted (2.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_sql_statistics_persisted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25863

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_sql_statistics_persisted [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9221732?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9221732?buildTab=artifacts#/) on release-23.1 @ [070410878098627863fe27afac3c60324e092f4a](https://github.com/cockroachdb/cockroach/commits/070410878098627863fe27afac3c60324e092f4a):


```
            │   │           └── • scan
            │   │                 columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id, crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, contention_time)
            │   │                 estimated row count: 0 (33% of the table; stats collected <hidden> ago)
            │   │                 table: statement_statistics@contention_time_idx (partial index)
            │   │                 spans: /2023-03-21T14:05:00.000001Z-
            │   │
            │   └── • index join
            │       │ columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, app_name, execution_count, service_latency, cpu_sql_nanos, contention_time, total_estimated_execution_time, p99_latency, metadata, statistics)
            │       │ estimated row count: 0
            │       │ table: statement_statistics@primary
            │       │ key columns: crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id
            │       │
            │       └── • top-k
            │           │ columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id, crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, total_estimated_execution_time)
            │           │ ordering: -total_estimated_execution_time
            │           │ estimated row count: 0
            │           │ order: -total_estimated_execution_time
            │           │ k: 500
            │           │
            │           └── • scan
            │                 columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id, crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, total_estimated_execution_time)
            │                 estimated row count: 0 (33% of the table; stats collected <hidden> ago)
            │                 table: statement_statistics@total_estimated_execution_time_idx (partial index)
            │                 spans: /2023-03-21T14:05:00.000001Z-
            │
            └── • index join
                │ columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, app_name, execution_count, service_latency, cpu_sql_nanos, contention_time, total_estimated_execution_time, p99_latency, metadata, statistics)
                │ estimated row count: 0
                │ table: statement_statistics@primary
                │ key columns: crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id
                │
                └── • top-k
                    │ columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id, crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, p99_latency)
                    │ ordering: -p99_latency
                    │ estimated row count: 0
                    │ order: -p99_latency
                    │ k: 500
                    │
                    └── • scan
                          columns: (aggregated_ts, fingerprint_id, transaction_fingerprint_id, plan_hash, app_name, node_id, crdb_internal_aggregated_ts_app_name_fingerprint_id_node_id_plan_hash_transaction_fingerprint_id_shard_8, p99_latency)
                          estimated row count: 0 (33% of the table; stats collected <hidden> ago)
                          table: statement_statistics@p99_latency_idx (partial index)
                          spans: /2023-03-21T14:05:00.000001Z-
[20:33:56] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12388/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted with config local: 6 tests, 1 failures
    logic.go:3942: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12388/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted:349: error while processing
    logic.go:3942: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12388/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted:349: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9986f69167dd2633633c749cd542ec4c/logTestExecBuild_sql_statistics_persisted2070579684
--- FAIL: TestExecBuild_sql_statistics_persisted (2.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_sql_statistics_persisted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted
+++ b/pkg/sql/opt/exec/execbuilder/testdata/sql_statistics_persisted
@@ -3,7 +3,7 @@
 statement ok
 CREATE STATISTICS system_statement_stats FROM system.statement_statistics
 
-query T
+query T retry
 EXPLAIN (VERBOSE)
 SELECT * FROM ((SELECT
   aggregated_ts,
@@ -478,7 +478,7 @@ vectorized: true
 statement ok
 CREATE STATISTICS system_transaction_stats FROM system.transaction_statistics
 
-query T
+query T retry
 EXPLAIN (VERBOSE)
 SELECT * FROM ((SELECT
   aggregated_ts,

